backend/src/routes/v0/memory.py

```
@@ -0,0 +1,105 @@
+    store: BaseStore = Depends(get_store),
+    limit: int = Query(default=10, ge=1, le=100),
+    offset: int = Query(default=0, ge=0),
+    query: str = Query(default=""),
+) -> MemoryListResponse:
+    try:
+        repo = _get_repo(user, store)
+        memories, total = await repo.list(limit=limit, offset=offset, query=query)
+        return MemoryListResponse(
+            memories=memories, total=total, limit=limit, offset=offset
+        )
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)